### PR TITLE
(test)O3-2111: Add patient chart add program e2e test

### PR DESCRIPTION
**Pull request by RandilaP**
## Requirements

- [ ] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Goal of this PR is to add a e2e test of add program function.

## Screenshots
image

## Related Issue

https://issues.openmrs.org/browse/O3-2111

### CODE REVIEW

**Comment by RandilaP** [comment]
Hey guys. Could you please take a look at this? I need some help with adding a data test id and I like to ask your opinion about what I'm missing @anjula-sack @Piumal1999 @jayasanka-sack

File: e2e/specs/add-program.spec.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { test } from '../core';
+import { ProgramPage } from '../pages';
+import { expect } from '@playwright/test';
+import { generateRandomPatient, deletePatient, Patient } from '../commands';
+
+let patient: Patient;
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
```
Comment (anjula-sack) [inline e2e/specs/add-program.spec.ts:10]:
```suggestion
});

```

File: e2e/specs/add-program.spec.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  });
+
+  await test.step('And I click on the Save button', async () => {
+    await programPage.page.getByText('Save').click();
+  });
+
+  await test.step('Then I should see the program in the patient chart', async () => {
+    await expect(programPage.tableRow().getByText('PMTCT')).toBeVisible();
+  });
+});
```
Comment (anjula-sack) [inline e2e/specs/add-program.spec.ts:37]:
```suggestion
});

```

File: packages/esm-patient-programs-app/src/programs/programs-overview.component.tsx [hunk 3]
```diff
@@ -159,7 +159,7 @@ const ProgramsOverview: React.FC<ProgramsOverviewProps> = ({ basePath, patientUu
                     ))}
                   </TableRow>
                 </TableHead>
-                <TableBody>
+                <TableBody data-testid="playwright1">
```
Comment (anjula-sack) [inline packages/esm-patient-programs-app/src/programs/programs-overview.component.tsx:162]: Let's remove the unwanted test ids @RandilaP

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by RandilaP** [comment]
@jayasanka-sack @anjula-sack @Piumal1999 I updated this PR with the current requirements that you asked for in previous PRs. I couldn't test this in the local server because of the issues with the form. But I tested this using the dev server. After testing it I added test ids and used them. I also locate each cells like in the allergies PR. I updated this test in to add a program and edit a program as @anjula requested

File: packages/esm-patient-programs-app/src/programs/programs-detailed-summary.component.tsx [hunk 4]
```diff
@@ -139,14 +139,14 @@ const ProgramsDetailedSummary: React.FC<ProgramsDetailedSummaryProps> = ({ patie
...
-                <TableBody>
+                <TableBody data-testid="programs-table">
                   {rows.map((row, i) => (
                     <TableRow key={row.id}>
                       {row.cells.map((cell) => (
                         <TableCell key={cell.id}>{cell.value?.content ?? cell.value}</TableCell>
                       ))}
-                      <TableCell className="cds--table-column-menu">
-                        <ProgramEditButton programEnrollmentId={enrollments[i]?.uuid} />
+                      <TableCell className="cds--table-column-menu" data-testid={'button'}>
```
Comment (anjula-sack) [inline packages/esm-patient-programs-app/src/programs/programs-detailed-summary.component.tsx:148]: Is this required? @RandilaP

**Comment by RandilaP** [comment]
@jayasanka-sack Could you please review this

File: e2e/specs/add-and-edit-program.spec.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+test.beforeEach(async ({ api }) => {
+  patient = await generateRandomPatient(api);
+});
+
+test('Should add a program and edit program', async ({ page, api }) => {
+  const programPage = new ProgramPage(page);
+  const row = programPage.table().locator('tr');
+  const programCell = row.locator('td:first-child');
+  const LocationCell = row.locator('td:nth-child(2)');
```
Comment (anjula-sack) [inline e2e/specs/add-and-edit-program.spec.ts:16]:
```suggestion
  const locationCell = row.locator('td:nth-child(2)');
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Review by anjula-sack — APPROVED** [review]
Good job! @RandilaP

**Pull request merged**